example-frontend/pages/autoOffsetExactOutToken.tsx

```diff
@@ -0,0 +1,73 @@
+import { useProvider, useSigner } from "wagmi";
+import { useState } from "react";
+import { ethers } from "ethers";
+import { FormatTypes, Interface, parseUnits } from "ethers/lib/utils";
+import OffsetHelper from "../../deployments/polygon/OffsetHelper.json";
+import ERC20 from "../abis/ERC20.json";
+
+export default function AutoOffsetExactOutToken() {
+  // input data for the autoOffsetExactOutToken function
+  const poolAddress = "0xD838290e877E0188a4A44700463419ED96c16107"; // Polygon - NCT
+  const depositedToken = "0x2791Bca1f2de4661ED88åA30C99A7a9449Aa84174"; // Polygon - USDC
+  const amount = parseUnits("0.0001", 6);
+  const [tx, setTx] = useState("");
+
+  // get signer & provider
+  const { data: signer } = useSigner();
+  const provider = useProvider();
+
+  // create contract for approve function of the ERC20 token
+  const iface = new Interface(ERC20.abi);
+  iface.format(FormatTypes.full);
+  const depositedTokenContract = new ethers.Contract(
+    depositedToken,
+    iface,
+    signer || provider
+  );
+
+  // create OffsetHelper contract
+  const offsetHelper = new ethers.Contract(
+    OffsetHelper.address,
+    OffsetHelper.abi,
+    signer || provider
+  );
+  const offset = async () => {
+    try {
+      // determine how much of the ERC20 token must be sent
+      const amountOut = await offsetHelper.calculateNeededTokenAmount(
+        depositedToken,
+        poolAddress,
+        amount
+      );
+
+      // approve spending of deposited tokens
+      await (
+        await depositedTokenContract.approve(offsetHelper.address, amountOut)
+      ).wait();
+
+      // retire carbon credits using ERC20 token, specifying the exact amount of TCO2s to retire,
+      const result = await offsetHelper.autoOffsetExactOutToken(
+        depositedToken,
+        poolAddress,
+        amount
+      );
+
+      setTx(result.hash);
+    } catch (error) {
```

Comment:
This still fails with the ENS error message
<img width="1143" alt="Xnapper-2023-10-07-16 47 14" src="https://github.com/ToucanProtocol/OffsetHelper/assets/9334296/e6acd8bf-e4bf-4055-9e5a-c1a780477951">

Reply:
fixed this.